src/NuGet.Core/NuGet.Packaging/Signing/Signatures/ISignature.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if IS_DESKTOP
+using System.Security.Cryptography.Pkcs;
+#endif
+
+namespace NuGet.Packaging.Signing
+{
+    public interface ISignature
+    {
+#if IS_DESKTOP
+        SignatureType Type { get; }
+
+        string FriendlyName { get; }
```

Comment:
Does `FriendlyName` need to be part of the interface?  It's internally used for error messages, but I'm not sure it should be part of the interface.